### PR TITLE
[ISSUE # 2233] Fixed crash when showing QR code

### PR DESCRIPTION
**Pull request by jeluard**
addresses #2233

### Summary:

Fixed crash when displaying public key QR code. Also improved labels show (not wallet address anymore).

### Steps to test:
- Open Status
- Navigate Jarrad's profile
- Display it's public key as QR code: it should work fine

[comment]: # (PRs will only be accepted if squashed into single commit.)

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/profile/qr_code/views.cljs [hunk 1]
```diff
@@ -35,15 +35,18 @@
                                                                             :height (.-height layout)}]))}
       (when (:width dimensions)
         [react/view {:style (styles/qr-code-container dimensions)}
-         [qr-code {:value (eip67/generate-uri (get contact qr-source) (when amount? {:value  amount}))
-                   :size  (- (min (:width dimensions)
-                                  (:height dimensions))
-                             80)}]])]
+         (when-let [value (eip67/generate-uri qr-value (when amount? {:value  amount}))]
```
Comment (yenda) [inline src/status_im/ui/screens/profile/qr_code/views.cljs:38]: why does this happen in the view ?

File: src/status_im/ui/screens/profile/views.cljs [hunk 2]
```diff
@@ -213,7 +214,7 @@
        [action-button {:label     (label :t/show-qr)
                        :icon      :icons/qr
                        :icon-opts {:color :blue}
-                       :on-press  (show-qr current-account :public-key)}]]
+                       :on-press  (show-qr current-account :public-key (:public-key current-account))}]]
```
Comment (yenda) [inline src/status_im/ui/screens/profile/views.cljs]: rather get the public-key in the destructuring ?

**Review by yenda — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/profile/qr_code/views.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/profile/qr_code/views.cljs:38]: What? `nil` value? It should not but if it does app crash so this is defensive.

**Comment by jeluard** [comment]
@yenda Pused fixes based on your feedback

File: src/status_im/ui/screens/profile/qr_code/views.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/screens/profile/qr_code/views.cljs:38]: No the generate-uri call, these things should be done in a subscription, views shouldn't contain any kind of computations

Comment (jeluard) [reply src/status_im/ui/screens/profile/qr_code/views.cljs:38]: That's not the scope of this PR so I would suggest to leave it out for now.

**Review by yenda — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Pull request merged**